### PR TITLE
Fix API oddity so that provider without subtypes can actually return empty array

### LINKED ISSUE #180: Rethink behavior in `Core_Sitemaps_Provider::get_object_subtypes()`
See discussion at https://github.com/GoogleChromeLabs/wp-sitemaps/pull/174/files#r423261977

### PR DESCRIPTION
**Pull request by felixarntz**
### Issue Number
Fixes #180 

### Description
Fixing this was very straightforward, we just needed to add a few lines of code to manually handle the empty case, which makes the code more understandable and allows a proper empty array return value for providers that don't support any object subtypes.

Tests still all pass, sitemaps and their names remain unchanged, so this should be good to go.

### Type of change
Please select the relevant options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement (change which improves an existing feature. E.g., performance improvement, docs update, etc.)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Acceptance criteria
- [x] My code follows WordPress coding standards.
- [x] I have performed a self-review of my own code.
- [ ] If the changes are visual, I have cross browser / device tested.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added test instructions that prove my fix is effective or that my feature works.

### CODE REVIEW

**Review by swissspidy — APPROVED** [review]
Neat!

File: inc/class-core-sitemaps-provider.php [hunk 1]
```diff
@@ -110,9 +110,20 @@ public function get_sitemap_type_data() {
...
+			);
+			return $sitemap_data;
+		}
+
+		// Otherwise, include individual sitemaps for every object subtype.
 		foreach ( $object_subtypes as $object_subtype_name => $data ) {
 			$sitemap_data[] = array(
-				'name'   => $object_subtype_name,
+				'name'  => $object_subtype_name,
 				'pages' => $this->max_num_pages( $object_subtype_name ),
```
Comment (swissspidy) [inline inc/class-core-sitemaps-provider.php]: Looks like `$object_subtype_name` needs to be explicitly cast to a string?

```
Parameter #1 $object_subtype of method                              
         Core_Sitemaps_Provider::max_num_pages() expects string, int|string  
         given. 
```

Comment (felixarntz) [reply inc/class-core-sitemaps-provider.php]: Where would this be an integer? This should always be a string, if it's not, we may need to fix something elsewhere.

Comment (swissspidy) [reply inc/class-core-sitemaps-provider.php]: The static analysis (PHPStan) doesn't know that `$object_subtypes` is an associative array keyed with strings.

Not sure why it wasn't flagged before. But `$this->max_num_pages( (string) $object_subtype_name )` should do the trick.

Comment (felixarntz) [reply inc/class-core-sitemaps-provider.php]: Not familiar with that tool, seems a bit odd to me if you can't configure it like that - or it's configured overly strict here, cause this is a false positive. But anyway, let's do the string cast then.

**Pull request merged**